src/CommunityToolkit.Maui.Core/Views/OnScreenSize/AppleScreenDensityHelper.shared.cs

```diff
@@ -0,0 +1,274 @@
+using System.Diagnostics.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.Core.Views.OnScreenSize;
+
+/// <summary>
+/// Helper class for retrieving PIxel per Inches (PPI) for apple devices so that the screen diagonal size in inches can be calculated later on. 
+/// </summary>
+/// <remarks>
+/// Because Apple does not provide any API for dynamically retrieving Pixel per Pnches (PPI) for all its devices (as Android does),
+/// we need to keep it hardcoded inside this class the PPI info based on device models,
+/// device names, and screen sizes. So that we can fallback to each one of the info (device name, model and screen size)
+/// to get PPI.ided apple-device-model.
+/// Probably we will need to update this list as new iphones and apple devices are launched having different PPI.  
+/// </remarks>
+[SuppressMessage("Style", "IDE0040:Add accessibility modifiers")]
+internal static class AppleScreenDensityHelper
+{
+
+	/// <summary>
+	/// Attempts to retrieve the pixel-per-inches (PPI) for the provided apple-device-model
+	/// </summary>
+	/// <param name="appleDeviceModel"></param>
+	/// <param name="ppi"></param>
+	/// <returns></returns>
+	internal static bool TryGetPpiByDeviceModel(string appleDeviceModel, out double ppi)
```

Comment:
If I put it inside *.macios.cs file, how can I unit test it?
Besides the name, it does not call any Apple specific platform code. 

Maybe you are going to say that I should move it out of core, but how can I call it from outside core (im talking about in terms of project dependencies here, not code itself). 

Maybe registering an interface on di? Probably not because the code itself does not have access to DI container for manually calling it.

Reply:
We test only shared.cs files.
If file has only code related to Apple it should not be visible/executable on other platforms.

Reply:
That's the case, it is *.shared.cs file. 

It was just put inside the the core because I needed to call it from within my *.ios.cs file. 

I must test it. I do not feel comfortable leaving this class without unit testing it. 

So que question remains the same: how to move it outside the core project while being able to call it from within the core project if core project cannot references mct.maui project (circular project dependencies issues)?